dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+                if (!locatorsFileContent.contains(locatorLogDir)) {
+                    locatorsFileContent.add(locatorLogDir);
+                    snappyTest.writeToFile(locatorLogDir, locatorsFile);
+                }
+            } else if (dirPath.contains("Store") || dirPath.contains("server")) {
+                String serverLogDir = localHost + " -dir=" + dirPath;
+                if (!serversFileContent.contains(serverLogDir)) {
+                    serversFileContent.add(serverLogDir);
+                    snappyTest.writeToFile(serverLogDir, serversFile);
+                }
+            } else if (dirPath.contains("lead")) {
+                String leadLogDir = localHost + " -dir=" + dirPath;
+                if (leadHost == null) {
+                    leadHost = localHost;
+                }
+                Log.getLogWriter().info("Lead host is: " + leadHost);
+                if (!leadsFileContent.contains(leadLogDir)) {
+                    leadsFileContent.add(leadLogDir);
+                    snappyTest.writeToFile(leadLogDir, leadsFile);
+                }
+            }
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getStackTrace());
+        }
+    }
+
+    protected void writeToFile(String logDir, File file) {
+        try {
+            FileWriter fw = new FileWriter(file.getAbsoluteFile(), true);
+            BufferedWriter bw = new BufferedWriter(fw);
+            bw.write(logDir);
+            bw.newLine();
+            bw.close();
+        } catch (IOException e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+    }
+
+    /**
+     * Executes user scripts for InitTask.
+     */
+    public static void HydraTask_executeSQLScriptsInInitTask() {
+        Vector scriptNames, paramList = null;
+        scriptNames = SnappyPrms.getSQLScriptNamesForInitTask();
+        paramList = SnappyPrms.getSQLScriptParamsForInitTask();
+        try {
+            for (int i = 0; i < scriptNames.size(); i++) {
+                String userScript = (String) scriptNames.elementAt(i);
+                String fileName = userScript;
+                String param = (String) paramList.elementAt(i);
+                String path = snappyTest.getParamPath(param);
+                String SnappyShellPath = snappyTest.getSnappyPath() + sep + "bin" + sep + "snappy-shell";
+                String filePath = snappyTest.getUserScriptLocation(fileName);
+                File file = new File(filePath);
+                File log = new File(".");
+                String dest = log.getCanonicalPath() + File.separator + "sqlScriptsInitTaskResult.log";
+                File logFile = new File(dest);
+                ProcessBuilder pb = new ProcessBuilder(SnappyShellPath, "run", "-file=" + file, "-param:path=" + path);
+                snappyTest.executeProcess(pb, logFile);
+            }
+        } catch (Exception e1) {
```

Comment:
The only exception here could be NPE. please look at the above comment and handle this properly. remove  try catch block.